scripts/postgresql-setup.sh

```diff
@@ -145,18 +153,20 @@ function dump_schema {
 
 function create_snapshot {
 	tgz_file=$1.tgz
-	dbfile=$1.sql
 	ledger_file=$2
 	tmp_dir=$(mktemp "${directory}" -t db-sync-snapshot-XXXXXXXXXX)
 	echo $"Working directory: ${tmp_dir}"
-	pg_dump --no-owner --schema=public "${PGDATABASE}" > "${tmp_dir}/$1.sql"
-	cp "$ledger_file" "$tmp_dir/$(basename "${ledger_file}")"
-	tar zcvf - --directory "${tmp_dir}" "${dbfile}" "$(basename "${ledger_file}")" | tee "${tgz_file}.tmp" \
-	| sha256sum | head -c 64 | sed -e "s/$/  ${tgz_file}\n/" > "${tgz_file}.sha256sum"
+	pg_dump --no-owner --schema=public --jobs="${numcores}" "${PGDATABASE}" --format=directory --file="${tmp_dir}/db/"
+	lstate_gz_file=$(basename "${ledger_file}").gz
+	gzip --to-stdout "${ledger_file}" > "${tmp_dir}/$(basename "${ledger_file}").gz"
+	tree "${tmp_dir}"
+	# Use plain tar here because the database dump files and the ledger state file are already gzipped.
+	tar cvf - --directory "${tmp_dir}" "db" "${lstate_gz_file}" | tee "${tgz_file}.tmp" \
+		| sha256sum | head -c 64 | sed -e "s/$/  ${tgz_file}\n/" > "${tgz_file}.sha256sum"
 	mv "${tgz_file}.tmp" "${tgz_file}"
 	rm "${recursive}" "${force}" "${tmp_dir}"
-	if test "$(gzip --test "${tgz_file}")" ; then
-	  echo "Gzip reports the snapshot file as being corrupt."
+	if test "$(tar --test --file "${tgz_file}")" ; then
```

Comment:
macOS tar does not have the `--test` flag and same for FreeBSD.

Reply:
:sob:

Reply:
Sorry I was dealing with Cardano FUD today instead of testing :(